<!-- MTracker simulator with manual controller launch file. -->
<launch>
  <param name="loop_rate" value="100"/>
  <param name="controls_topic" type="string" value="controls"/>
  <param name="virtual_pose_topic" type="string" value="virtual_pose"/>
  <param name="virtual_velocity_topic" type="string" value="virtual_velocity"/>
  <param name="reference_pose_topic" type="string" value="reference_pose"/>
  <param name="reference_velocity_topic" type="string" value="reference_velocity"/>

  <node name="reference_generator" pkg="mtracker" type="reference_generator" respawn="false" required="true" output="screen">
    <param name="trajectory_type" value="0"/>
    <param name="trajectory_paused" type="bool" value="false"/>
  </node>
  
  <node name="automatic_controller" pkg="mtracker" type="automatic_controller" respawn="false" required="true" output="screen">
  </node>

  <node name="simulator" pkg="mtracker" type="simulator" respawn="false" required="true" output="screen">
    <param name="time_constant" value="0.1"/>
  </node>

  <node name="rviz" pkg="rviz" type="rviz" respawn="false" required="true" output="screen" args="-d $(find mtracker)/resources/mtracker.rviz"/>
</launch>
